src/domdiv/config_options.py

```diff
@@ -834,6 +834,25 @@ def parse_opts(cmdline_args=None):
     return options
 
 
+def flatten_lower_and_deduplicate(l: any) -> set:
+    out = set()
+    if l is None:
+        return out
+    for item in l:
+        if isinstance(item, list):
+            # The arg parser sometimes generates a list of lists
+            out.update(flatten_lower_and_deduplicate(item))
+        elif isinstance(item, set):
+            out.update(item)
+        elif isinstance(item, str):
+            out.add(item.lower())
+        else:
+            logger.warning(
+                f"Ignoring configuration option {item} with type {type(item)} as it has an unexpected type."
+            )
```

Comment:
I feel like this should either add the item (so work for things that aren't strings) or throw an exception here?

Reply:
Updating it to throw an exception.